internal/provisioner/kops_provisioner_cluster.go

```diff
@@ -865,3 +875,87 @@ func (provisioner *KopsProvisioner) RefreshKopsMetadata(cluster *model.Cluster)
 
 	return nil
 }
+
+func cleanupMetricsServerResources(provisioner *KopsProvisioner, logger log.FieldLogger, k8sClient *k8s.KubeClient, cluster *model.Cluster) error {
+	ctx := context.TODO()
+	kops, err := kops.New(provisioner.s3StateStore, logger)
+	if err != nil {
+		return errors.Wrap(err, "failed to create kops wrapper")
+	}
+	defer kops.Close()
+
+	err = kops.ExportKubecfg(cluster.ProvisionerMetadataKops.Name)
+	if err != nil {
+		return errors.Wrap(err, "failed to export kubecfg")
+	}
```

Comment:
It looks like this kops object is created, but never really used for anything. The k8sclient we pass in already is configured to access the cluster.